rush-plugins/rush-serve-plugin/src/types.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import type { RushConfiguration, RushSession, IPhasedCommand } from '@rushstack/rush-sdk';
+
+export interface IPhasedCommandHandlerOptions {
+  rushSession: RushSession;
+  rushConfiguration: RushConfiguration;
+  command: IPhasedCommand;
+  portParameterLongName: string | undefined;
+  logServePath: string | undefined;
+  globalRoutingRules: IRoutingRule[];
+  buildStatusWebSocketPath: string | undefined;
+}
+export interface IRushProjectServeJson {
+  routing: IRoutingRuleJson[];
+}
+
+export interface IBaseRoutingRuleJson {
+  servePath: string;
+  immutable?: boolean;
+}
+
+export interface IRoutingFolderRuleJson extends IBaseRoutingRuleJson {
+  projectRelativeFile: undefined;
+  projectRelativeFolder: string;
+}
+
+export interface IRoutingFileRuleJson extends IBaseRoutingRuleJson {
+  projectRelativeFile: string;
+  projectRelativeFolder: undefined;
+}
```

Comment:
```suggestion
export interface IRoutingFolderRuleJson extends IBaseRoutingRuleJson {
  projectRelativeFile: never;
  projectRelativeFolder: string;
}

export interface IRoutingFileRuleJson extends IBaseRoutingRuleJson {
  projectRelativeFile: string;
  projectRelativeFolder: never;
}
```

Reply:
Setting to never makes some of the processing annoying.